### ISSUE TITLE
SMAC generates unnecessary warning

### ISSUE THREAD

**Issue by mfeurer** [automl/SMAC3#275]
in case the acquisition function value becomes `NaN` for some reason:

```
/usr/local/lib/python3.5/dist-packages/smac/optimizer/acquisition.py:164: RuntimeWarning: invalid value encountered in less
  if (f < 0).any():
```
Simply replacing `NaN` by zero would alleviate this problem. 

Also, we should maybe catch these kind of warnings and only show them if the logging level is set to debug:

```
/usr/local/lib/python3.5/dist-packages/scipy/stats/_distn_infrastructure.py:876: RuntimeWarning: invalid value encountered in less_equal
  return (self.a <= x) & (x <= self.b)
```

These two seem to confuse people unfamiliar with Bayesian optimization.

**Comment by mlindauer** [comment]
> in case the acquisition function value becomes NaN for some reason:

shouldn't we figure out the reason and try to fix this?
I worry a bit that there could be real issue hidden somewhere.

In principle, I agree that we should hide warnings that we cannot fix and that are acceptable for us.
In fact, we are doing this already for two warnings in the censored data imputation.

**Comment by mfeurer** [comment]
This should be fixed by #275.
